front/app/resources/GetSerieFromStream.tsx

```diff
@@ -154,9 +150,13 @@ export default class GetSerieFromStream extends PureComponent<Props, State> {
   }
 
   render() {
-    const { children } = this.props;
+    const { children, convertToGraphFormat } = this.props;
+    const { unconvertedSerie } = this.state;
+
     return (children as children)({
-      ...this.state,
+      serie: isNilOrError(unconvertedSerie)
+        ? unconvertedSerie
+        : convertToGraphFormat(unconvertedSerie),
```

Comment:
Instead of converting the serie in the `.subscribe` method of the observable above, I am now doing it just before passing the data down. The problem with the previous way of doing things was that `convertToGraphFormat` sometimes changes which should retrigger the resource component, but that wasn't happening. This way it works correctly